watson_developer_cloud/watson_service.py

```diff
@@ -259,6 +265,8 @@ def __init__(self, vcap_services_name, url, username=None, password=None,
         elif iam_access_token is not None or iam_apikey is not None:
             self.set_token_manager(iam_apikey, iam_access_token, iam_url)
 
+        self.load_from_credential_file(vcap_services_name)
```

Comment:
Based on the team discussion yesterday morning, we decided to go with what's essentially the display name, e.g. Visual Recognition. You'll need to get that value and search for the snake-cased form (`visual_recognition`) in your `_set_credential_based_on_type` function.